asapdiscovery-ml/asapdiscovery/ml/schema_v2/config.py

```diff
@@ -0,0 +1,1252 @@
+            Full ML dataset to split
+
+        Returns
+        -------
+        torch.nn.Subset
+            Train split
+        torch.nn.Subset
+            Val split
+        torch.nn.Subset
+            Test split
+        """
+        split_fracs = [self.train_frac, self.val_frac, self.test_frac]
+        # Check that split_fracs adds to 1, padding if it doesn't
+        # Add an allowance for floating point inaccuracies
+        total_splits = sum(split_fracs)
+        if not np.isclose(total_splits, 1):
+            sink_frac = 1 - total_splits
+            split_fracs = split_fracs[:2] + [sink_frac] + split_fracs[2:]
+            sink_split = True
+        else:
+            sink_split = False
+
+        # Calculate how many molecules we want covered through each split
+        n_mols_split = np.floor(np.asarray(split_fracs) * len(ds))
+
+        # First get all the unique created dates
+        dates_dict = {}
+        # If we have a grouped dataset, we want to iterate through compound_ids, which will
+        #  allow us to access a group of structures. Otherwise, loop through the structures
+        #  directly
+        if self.grouped:
+            iter_list = ds.compound_ids
+        else:
+            iter_list = ds.structures
+        for i, iter_item in enumerate(iter_list):
+            if self.grouped:
+                # Take the earliest date from all structures (they should all be the same,
+                #  but just in case)
+                all_dates = [
+                    s["date_created"]
+                    for s in ds.structures[iter_item]
+                    if "date_created" in s
+                ]
+                if len(all_dates) == 0:
+                    raise ValueError("Dataset doesn't contain dates.")
+                else:
+                    date_created = min(all_dates)
+            else:
+                try:
+                    date_created = iter_item["date_created"]
+                except KeyError:
+                    raise ValueError("Dataset doesn't contain dates.")
+            try:
+                dates_dict[date_created].append(i)
+            except KeyError:
+                dates_dict[date_created] = [i]
+        all_dates = np.asarray(list(dates_dict.keys()))
+
+        # Sort the dates
+        all_dates_sorted = sorted(all_dates)
+
+        # Make subsets
+        idx_lists = [dates_dict[d] for d in all_dates_sorted]
+        all_subsets = DatasetSplitterConfig._make_subsets(ds, idx_lists, n_mols_split)
+
+        # Take out the sink split
+        if sink_split:
+            ds_train, ds_val, _, ds_test = all_subsets
+        else:
+            ds_train, ds_val, ds_test = all_subsets
+
+        return ds_train, ds_val, ds_test
+
+
+class ModelType(str, Enum):
+    """
+    Enum for model types.
+    """
+
+    gat = "gat"
+    schnet = "schnet"
+    e3nn = "e3nn"
+    INVALID = "INVALID"
+
+
+class MTENNStrategy(str, Enum):
```

Comment:
Is there an argument that some of these should live inside MTENN rather than here? These are MTENN specific and could be of use there also? / We don't want to shadow the whole api here. Up to you.

Reply:
I've moved all the `Model` config stuff over to `mtenn`, since I agree that it probably fits better there. we can discuss at the dev sync today though